### PR TITLE
feat: support callback options for put and multipartUpload

### PR DESCRIPTION
**Pull request by binghaiwang**
1.support callback options for put and multipartUpload

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -793,29 +793,21 @@ parameters:
   - [mime] {String} custom mime, will send with `Content-Type` entity header
   - [meta] {Object} user meta, will send with `x-oss-meta-` prefix string
     e.g.: `{ uid: 123, pid: 110 }`
+  - [callback] {Object} The callback parameter is composed of a JSON string encoded in Base64,detail [see](https://www.alibabacloud.com/help/doc-detail/31989.htm)<br> 
+    - callbackUrl {String} After a file is uploaded successfully, the OSS sends a callback request to this URL. 
+    - [callbackHost] {String} The host header value for initiating callback requests.
+    - callbackBody {String} The value of the request body when a callback is initiated, for example, key=$(key)&etag=$(etag)&my_var=$(x:my_var). 
+    - [callbackBodyType] {String} The Content-Type of the callback requests initiatiated, It supports application/x-www-form-urlencoded and application/json, and the former is the default value.  
+  - [callbackVar] {Object} Custom parameters are a map of key-values<br>
```
Comment (PeterRao) [inline README.md]: 合并callbackVar到 callback里面

File: README.md [hunk 2]
```diff
@@ -793,29 +793,21 @@ parameters:
   - [mime] {String} custom mime, will send with `Content-Type` entity header
   - [meta] {Object} user meta, will send with `x-oss-meta-` prefix string
     e.g.: `{ uid: 123, pid: 110 }`
+  - [callback] {Object} The callback parameter is composed of a JSON string encoded in Base64,detail [see](https://www.alibabacloud.com/help/doc-detail/31989.htm)<br>
```
Comment (PeterRao) [inline README.md]: callback里面参数省略 callback 前缀

File: test/browser.tests.js [hunk 3]
```diff
@@ -17,15 +17,16 @@ var ossConfig;
...
 describe('browser', function () {
   before(function* () {
     ossConfig = {
       region: stsConfig.region,
       accessKeyId: stsConfig.Credentials.AccessKeyId,
       accessKeySecret: stsConfig.Credentials.AccessKeySecret,
       stsToken: stsConfig.Credentials.SecurityToken,
-      bucket: stsConfig.bucket
+      bucket: stsConfig.bucket,
+      callbackServer: 'http://oss-demo.aliyuncs.com:23450' //oss demo callback server an ecs, used with PutObject、PostObject、CompleteMultipartUpload
```
Comment (PeterRao) [inline test/browser.tests.js]: callbackServer 最为配置变量移到别处，会给开发者造成困惑。

File: README.md (hunk 1 above)
Comment (binghaiwang) [reply README.md]: fixed

File: README.md (hunk 2 above)
Comment (binghaiwang) [reply README.md]: fixed

File: test/browser.tests.js (hunk 3 above)
Comment (binghaiwang) [reply test/browser.tests.js]: fixed

**Pull request merged**